apps/full-site-editing/full-site-editing-plugin/full-site-editing/blocks/site-title/edit.js

```diff
@@ -0,0 +1,115 @@
+/* eslint-disable wpcalypso/jsx-classname-namespace */
+/**
+ * External dependencies
+ */
+import React, { Component } from 'react';
+import classNames from 'classnames';
+
+/**
+ * WordPress dependencies
+ */
+import { __ } from '@wordpress/i18n';
+import { withNotices, ServerSideRender } from '@wordpress/components';
+import { PlainText } from '@wordpress/block-editor';
+import apiFetch from '@wordpress/api-fetch';
+import { withSelect } from '@wordpress/data';
+import { compose } from '@wordpress/compose';
+
+/**
+ * Internal dependencies
+ */
+
+class SiteTitleEdit extends Component {
+	state = {
+		title: __( 'Site title loading...' ),
+		initialTitle: '',
+	};
+
+	componentDidMount() {
+		this.requestSiteTitle();
+	}
+
+	componentDidUpdate( prevProps ) {
+		const { title, initialTitle } = this.state;
+		const {
+			isSaving,
+			isPublished,
+			isPublishing,
+			isAutosaving,
+			noticeOperations,
+			isSelected,
+		} = this.props;
+
+		const titleUnchanged = title && title.trim() === initialTitle.trim();
+		const titleIsEmpty = ! title || title.trim().length === 0;
+
+		// Reset to initial title if user de-selects the block with an empty value
+		if ( ! isSelected && prevProps.isSelected && titleIsEmpty ) {
+			this.setSiteTitle( initialTitle );
+		}
+
+		// Don't do anything further if the title is unchanged or the saving state is the same as previous update
+		if ( titleUnchanged || prevProps.isSaving === this.props.isSaving ) {
+			return;
+		}
+
+		const userInitiatedPublish = ( ( isSaving && isPublished ) || isPublishing ) && ! isAutosaving;
+
+		// Save the title on publish
+		if ( userInitiatedPublish ) {
+			const { title } = this.state;
+
+			apiFetch( { path: '/wp/v2/settings', method: 'POST', data: { title } } )
+				.then( () => this.resetTitle() )
+				.catch( ( { message } ) => noticeOperations.createErrorNotice( message ) );
+		}
+	}
+
+	requestSiteTitle = () => {
+		const { noticeOperations } = this.props;
+
+		return apiFetch( { path: '/wp/v2/settings' } )
+			.then( ( { title } ) => this.setState( { initialTitle: title, title } ) )
+			.catch( ( { message } ) => noticeOperations.createErrorNotice( message ) );
+	};
+
+	setSiteTitle = value => this.setState( { title: value } );
+
+	resetTitle = () => this.setState( { initalTitle: this.state.title } );
+
+	render() {
+		const { title } = this.state;
+		const { isSelected, className } = this.props;
+
+		if ( isSelected ) {
+			return (
+				<h1>
+					<PlainText
+						className={ className }
+						value={ title }
+						onChange={ this.setSiteTitle }
```

Comment:
We need to set a block attribute here with `setAttribute` in order to let the editor know that the content has changed. Otherwise the "Update" button will remain disabled until we change the content of other blocks:

![Jun-13-2019 14-50-41](https://user-images.githubusercontent.com/1233880/59407007-e972f680-8dea-11e9-859c-58b0a41568b2.gif)

Reply:
I've got this working with attributes instead of local state for the value tracking. I admit I'm weirded out by the code view though; I really want whatever I see in the code view to coincide with what would get saved. If we never save attributes, should user ever see them in the code view?

<img width="903" alt="Screen Shot 2019-06-13 at 2 54 33 PM" src="https://user-images.githubusercontent.com/349751/59470068-77e77800-8deb-11e9-888d-d5da37267d0f.png">

Reply:
Actually, the code view is showing the content that is saved: nothing 🙂(`save: () => null`).

Remember that this is a dynamic block, so it is rendered by the server on the published view. Think of them like shortcodes. Wouldn't a shortcode allowing to override the site description look similar to that block syntax?

I admit the attribute is not strictly required when the post is saved (the back-end rendering will ignore the attribute and will retrieve always the description from the site data), but I don't know other way of letting the editor know there are changes that need to be saved.

Reply:
Let's tackle this next sprint in https://github.com/Automattic/wp-calypso/issues/34004